### ISSUE TITLE
What to do with Rubinius specs

### ISSUE THREAD

**Issue by headius** [ruby/rubyspec#62]
RubySpec never really died, it just split into two worlds: the ruby/rubyspec world, managed by ruby-core and JRuby committers; and the rubinius world. Both worlds have continued to add specs. Only the ruby/rubyspec world has committed to staying compatible (or rather, only that world has the keys to Ruby's future).

I'm wondering what if anything we should do with specs the Rubinius folks continue to add to their repo.

* Should we reapply the same commits? I have some concern here because they've stated that they no longer consider MRI to be the spec, so there could be differences of opinion on behavior.
* Should we pick and choose individual examples to copy over? This would be slower and lose lineage of the code, but it would allow us to only merge over what's appropriate.

There's a third option...we just ignore their specs. I think it would be a shame to lose them, though.

It might be possible to convince some of the Rubinius committers to submit their specs to ruby/rubyspec.

**Comment by headius** [comment]
The specs that prompted me to file this issue are recent additions for prepend. They *may* be mostly covered by MRI's tests, but I have not audited them in detail yet.

**Comment by eregon** [comment]
Yeah, let's ask rubinius/rubinius!

**Comment by eregon** [comment]
Per https://github.com/rubinius/rubinius/issues/3403, we could try importing their new specs if they are compatible enough, or just cherry-pick if not.
I intend to have a look in a few days.
Do you have other special interests beyond `Module#prepend` and related?

**Comment by headius** [comment]
Those were the specs I noticed most recently. Cherry-picking or hand
copying is certainly sufficient. It is too bad they don't want to collab on
the new rubyspec, but yay for OSS.

**Comment by eregon** [comment]
I merged most specs (68 commits) from the rubinius repository.
I reviewed quickly all of them and skipped a few bad specs.

It is not really trivial as both repositories do not a share a common ancestor.

The starting point (last common commit) was:
`Added specs for CAPI function rb_cloexec_open`
rubinius/rubinius@94a7ce1b85db4c57438e56b67acb36d912021cae
rubyspec/rubyspec@671ef6d32637fef783e82c129a44ce10063718b0
For the filter-branch repo, this is commit
`3773a700a909aa9f799b2a5cb3f0f252f7068aba`.

For the procedure, it requires some git-fu but in the end filter-branch + rebase did it.
```bash
# In a copy of rubinius repo
git checkout -b rubyspec
git filter-branch --subdirectory-filter spec/ruby 
git push ../rubyspec rubyspec:rbx

cd ../rubyspec
git checkout rbx
git checkout -b rbx_rebased
# The commit hash is from the filtered repo for the last common point we merged
git rebase --onto master 3773a700a9 rbx_try
# Have some fun and --skip if the patch is too bad
```

**Comment by eregon** [comment]
The merge point for next time is
`39be273c892e1c839d119587882d149902260ab3`
in the filtered branch.

**Comment by eregon** [comment]
I merged until rubinius/rubinius@291c7c7d07114e3787465418fb56bae841802456 on June 26.

**Comment by eregon** [comment]
I merged until https://github.com/rubinius/rubinius/commit/9aa7833c1a0a2e3cd2a363441eb18b97489f182d on July 27.

**Comment by eregon** [comment]
I merged until rubinius/rubinius@dbd33a6dadd407041fcafa6651986ae02551f3ee on August 28.

**Comment by eregon** [comment]
I merged specs in the [2.2 branch](https://github.com/rubinius/rubinius/commits/2.2/spec/ruby) until https://github.com/rubinius/rubinius/commit/da448d50480d01de272ff222267ebc626bf4bd07 on September 29.

**Comment by eregon** [comment]
I merged https://github.com/rubinius/rubinius/commit/fcf450d47a33aed9f9cebc58ede42df92b340b6e from master and until https://github.com/rubinius/rubinius/commit/649739c34bcda5020fd0da52c8a3e60137f5da9e on the 2.2 branch.
